base/set.jl

```diff
@@ -1,5 +1,35 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+	Set{T} <: AbstractSet{T}
+
+`Set`s are containers that provide fast membership testing with `in`.
+
+`Set`s have efficient implementations of set operations such as `union` and `intersect`.
+Elements in a `Set` are unique, as determined by the elements' definitions of
+`isequal` and `hash`.
+The order of elements in a `Set` is an implementation detail and cannot be relied on.
+
+See also: [`AbstractSet`](@ref), [`BitSet`](@ref), [`Dict`](@ref),
+[`push!`](@ref), [`empty!`](@ref), [`union!`](@ref), [`in`](@ref)
+
+# Examples
+```jldoctest filter = r"^Set{[A-Za-z0-9]+} with [0-9]+ element(s)?:"
+julia> s = Set(Any[0, -0.0, 0.0])
+Set{Any} with 2 elements:
+  0.0
+  -0.0
+
+julia> push!(s, 'a'); push!(s, 0.0)
```

Comment:
What about pushing `0` to better highlight the mutability?
```suggestion
julia> push!(s, 'a'); push!(s, 0)
```
... and than `0.0 in s`

Reply:
The issue with that is that I think the current behaviour where `0` overwrites `0.0` is an implementation detail (i.e. I think keeping `0.0` would be documented behaviour, too), so it's better to not have it in the doc example.